ansible_base/authentication/models/authenticator_map.py

```diff
@@ -27,46 +27,56 @@ class Meta:
         Authenticator,
         null=False,
         on_delete=models.CASCADE,
-        help_text=(_("The authenticator this mapping belongs to")),
+        help_text=_("The authenticator this mapping belongs to"),
         related_name="authenticator_maps",
     )
     revoke = models.BooleanField(
         null=False,
         default=False,
-        help_text=(_("If a user does not meet this rule should we revoke the permission")),
+        help_text=_("If a user does not meet this rule should we revoke the permission"),
     )
+
     map_type = models.CharField(
         max_length=17,
         null=False,
         default="team",
         choices=[
-            ('team', 'team'),
-            ('is_superuser', 'is_superuser'),
-            ('is_system_auditor', 'is_system_auditor'),
             ('allow', 'allow'),
+            ('is_superuser', 'is_superuser'),
+            ('role', 'role'),
             ('organization', 'organization'),
+            ('team', 'team'),
         ],
-        help_text=(_('What does the map work on, a team, a user flag or is this an allow rule')),
+        help_text=_('What will the map grant the user? System access (allow) a team or organization membership, the superuser flag or a role in the system'),
+    )
+
+    role = models.CharField(
+        max_length=512,
+        null=True,
```

Comment:
SonarCloud doesn't like this `null=True` flag because it doesn't make sense for a `CharField`. I've seen this before and I think it's very valid, although I accept that other existing fields may do the same thing.

Trying to remember the academic issue: Null isn't actually a value in the database, which is what this is about. If you want serializers to accept null or blank values, that's a separate topic, and not what _this_ means.

Reply:
I want to be compatible with the other char fields representing "foreign keys by name". So it makes sense to either have `null=True` for all or none of them